manifests/config.pp

```diff
@@ -0,0 +1,190 @@
+#   A different user to run commands as after login.
+#
+# @param log_console_level
+#   The type of information to display on the console. Valid options are debug, info, notice,
+#   warn, and error. The default is info.
+#
+# @param log_file
+#   The path and name of the log file.
+#
+# @param log_file_level
+#   The type of information to record in the log file. Valid options are debug, info, notice,
+#   warn, and error. By default in Bolt this is info.
+#
+# @param log_file_append
+#   Add output to an existing log file. By default in Bolt this is true.
+#
+# @param disable_analytics
+#   A different user to run commands as after login.
+#
+class simp_bolt::config (
+  String                                               $local_user         = $simp_bolt::user::username,
+  String                                               $local_group        = $simp_bolt::user::username,
+  Pattern['^/']                                        $local_user_home    = $simp_bolt::user::home,
+  Optional[String]                                     $modulepath         = undef,
+  Optional[Boolean]                                    $color              = undef,
+  Optional[Integer[0]]                                 $concurrency        = undef,
+  Optional[Enum['human','json']]                       $format             = undef,
+  Optional[String]                                     $hiera_config       = undef,
+  Optional[String]                                     $inventoryfile      = undef,
+  Optional[Enum['docker','local','pcp','ssh','winrm']] $transport          = undef,
+  Optional[Boolean]                                    $ssh_host_key_check = undef,
+  Optional[String]                                     $ssh_password       = undef,
+  Optional[Integer[0]]                                 $ssh_port           = undef,
+  Optional[String]                                     $ssh_private_key    = undef,
+  Optional[String]                                     $ssh_proxyjump      = undef,
+  String                                               $ssh_tmpdir         = $simp_bolt::user::home,
+  String                                               $ssh_user           = $simp_bolt::user::username,
+  String                                               $ssh_run_as         = 'root',
+  Enum['debug', 'info', 'notice', 'warn', 'error']     $log_console_level  = 'info',
+  String                                               $log_file           = '/var/log/puppetlabs/bolt/bolt.log',
+  Enum['debug', 'info', 'notice', 'warn', 'error']     $log_file_level     = 'info',
+  Optional[Boolean]                                    $log_file_append    = undef,
+  Boolean                                              $disable_analytics  = true,
+){
+  assert_private()
+
+# Define and create the Boltdir and its parent directory
+  $_bolt_dir_parent = "${local_user_home}/.puppetlabs"
+  $_bolt_dir = "${local_user_home}/.puppetlabs/bolt"
+
+  file { $_bolt_dir_parent:
```

Comment:
You should probably create the `local_user_home` here if you don't include the `bolt::user` class.

Since this is  a `private` class, this can all be orchestrated via the `init.pp`

Reply:
Changed to create the home directory if it does not exist.